dandiapi/api/views/asset.py

```diff
@@ -194,20 +199,26 @@ def update(self, request, versions__dandiset__pk, versions__version, **kwargs):
                 status=status.HTTP_405_METHOD_NOT_ALLOWED,
             )
 
-        serializer = AssetRequestSerializer(data=request.data)
+        serializer = AssetUpdateRequestSerializer(data=request.data)
         serializer.is_valid(raise_exception=True)
 
-        asset_blob = get_object_or_404(AssetBlob, blob_id=serializer.validated_data['blob_id'])
-
-        metadata = serializer.validated_data['metadata']
-        if 'path' not in metadata:
-            return Response('No path specified in metadata', status=404)
-        path = metadata['path']
-        # Strip away any computed fields
-        metadata = Asset.strip_metadata(metadata)
-        asset_metadata, created = AssetMetadata.objects.get_or_create(metadata=metadata)
-        if created:
-            asset_metadata.save()
+        if 'blob_id' in serializer.validated_data:
+            asset_blob = get_object_or_404(AssetBlob, blob_id=serializer.validated_data['blob_id'])
+        else:
+            asset_blob = old_asset.blob
+
+        if 'metadata' in serializer.validated_data:
+            metadata = serializer.validated_data['metadata']
+            if 'path' not in metadata:
+                return Response('No path specified in metadata', status=404)
+            path = metadata['path']
+            # Strip away any computed fields
```

Comment:
before stripping - could we test that if dandi-etag is provided among digests, it matches the one of the `asset_blob`?

If not matches, it means that metadata is "wrong" in that it is not applicable to the blob/asset, and thus better not be allowed to be uploaded

Reply:
Since the billet of this PR (and #392, which it closes) is pretty narrow, just focusing on unblocking a specific problem of non-optional parameters, I'd like to defer this proposed functionality to a new issue. @yarikoptic, do you mind opening such an issue?